internal/outputs/infra.go

```diff
@@ -6,47 +6,49 @@
 package outputs
 
 import (
+	"fmt"
 	"os"
 
-	"github.com/newrelic/nri-flex/internal/load"
-	"github.com/sirupsen/logrus"
-
 	Integration "github.com/newrelic/infra-integrations-sdk/integration"
+	"github.com/newrelic/nri-flex/internal/load"
 )
 
 // InfraIntegration Creates Infrastructure SDK Integration
-func InfraIntegration() {
-	load.Hostname, _ = os.Hostname() // set hostname
-
+func InfraIntegration() error {
 	var err error
+	load.Hostname, err = os.Hostname() // set hostname
+	if err != nil {
+		load.Logrus.
+			WithError(err).
+			Debug("flex: failed to get the hostname while creating integration")
+	}
+
 	load.Integration, err = Integration.New(load.IntegrationName, load.IntegrationVersion, Integration.Args(&load.Args))
+	if err != nil {
+		return fmt.Errorf("flex: failed to create integration %v", err)
+	}
 
 	// Accepts ConfigPath as alias for ConfigFile. This will allow the Infrastructure Agent
 	// passing an embedded config via the default CONFIG_PATH environment variable
 	if load.Args.ConfigPath != "" {
 		load.Args.ConfigFile = load.Args.ConfigPath
 	}
 
+	load.Entity, err = createEntity(load.Args.Local, load.Args.Entity)
 	if err != nil {
-		load.Logrus.WithFields(logrus.Fields{"err": err}).Fatal("flex: create integration")
-	}
-
-	if load.Args.Local {
-		load.Entity = load.Integration.LocalEntity()
-	} else {
-		InfraRemoteEntity()
+		return fmt.Errorf("flex: failed create remote entity: %v", err)
```

Comment:
in this case it should be just "failed to create entity", because it can be a local or remote entity, right?